cli/azd/cmd/login.go

```diff
@@ -70,7 +70,10 @@ func (lf *loginFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandO
 	local.BoolVar(
 		&lf.useDeviceCode,
 		"use-device-code",
-		false,
+		// For Codespaces in VSCode Browser, interactive browser login will 404 when attempting to redirect to localhost
+		// (since azd cannot launch a localhost server when running remotely).
+		// Hence, we default to device-code. See https://github.com/Azure/azure-dev/issues/1006
+		os.Getenv("CODESPACES") == "true",
```

Comment:
Last I implemented, we needed this for remote containers as well.  It was removed here:  https://github.com/Azure/azure-dev/commit/46515fe95249399f1f8deca8aec79a67928b738e

Reply:
@jongio Note that we've found recently that device code prevents the user from accessing MSGraph calls, which prevents the user from running `azd pipeline config`. Thus, defaulting to device code isn't always a benefit.

For Devcontainer, auth code / interactive login will work if it's running on local WSL / local docker. It will not work on remote sessions. I'd like not to default in that case as a result. They're advanced users that will answer what to do to configure remote.

For Codespaces, the audience can be someone that is just trying `azd`, so I've decided to make the change for there.